src/renderer/base/renderer.cpp

```diff
@@ -1231,13 +1254,38 @@ void Renderer::_PaintSelection(_In_ IRenderEngine* const pEngine)
                 }
             }
         }
+    }
+    CATCH_LOG();
+}
+
+// Routine Description:
+// - returns search highlighted areas within the dirty regions of the window
+SearchHighlights Renderer::_GetDirtySearchHighlights(IRenderEngine* const pEngine) const
+{
+    std::span<const til::rect> dirtyAreas;
+    LOG_IF_FAILED(pEngine->GetDirtyArea(dirtyAreas));
 
-        if (!dirtySearchRectangles.empty())
+    SearchHighlights dirtySearchRects;
+    for (const auto& dirtyRect : dirtyAreas)
+    {
+        for (const auto& sr : _currentSearchHighlights.all)
         {
-            LOG_IF_FAILED(pEngine->PaintSelections(std::move(dirtySearchRectangles)));

```

Comment:
note to self: okay so this still works in conhost, in GDI, same as before, because GDI is still just using the selection rect from `_GetSelectionRects`, and painting in the PaintSelection call on L1253